src/graph/backend/dnnl/kernels/sdp_decomp_config.cpp

```diff
@@ -822,6 +822,32 @@ dnnl::primitive_attr sdp_decomp_config_t::make_primitive_attr(
     return attr;
 }
 
+#define DECLARE_RESET_ENGINE(primitive_name, primitive_type) \
+    { \
+        const auto desc_t = (primitive_name).get_primitive_desc()->impl(); \
+        dnnl_primitive_desc new_pd_t(desc_t, p_engine.get()); \
+        primitive_type::primitive_desc new_pd(&new_pd_t); \
+        (primitive_name) = primitive_type(new_pd); \
+    }
+
+impl::status_t sdp_decomp_config_t::reset_engine(const dnnl::engine &p_engine) {
+#if DNNL_CPU_RUNTIME == DNNL_RUNTIME_OMP
+    omp_set_num_threads(1);
```

Comment:
why to set thread num as 1? If we want to avoid race conditions caused by multithreading, we should use locking instead of changing the number of threads inside the function.